soh/soh/Enhancements/AssignableTunicsAndBoots.cpp

```diff
@@ -0,0 +1,108 @@
+    ) {
+        return;
+    }
+
+    s32 i;
+    for (i = 0; i < ARRAY_COUNT(sItemButtons); i++) {
+        if (CHECK_BTN_ALL(input->press.button, sItemButtons[i])) {
+            break;
+        }
+    }
+    s32 item = Player_GetItemOnButton(play, i);
+    if (item >= ITEM_TUNIC_KOKIRI && item <= ITEM_BOOTS_HOVER) {
+        if (item >= ITEM_BOOTS_KOKIRI) {
+            u16 bootsValue = item - ITEM_BOOTS_KOKIRI + 1;
+            if (CUR_EQUIP_VALUE(EQUIP_TYPE_BOOTS) == bootsValue) {
+                Inventory_ChangeEquipment(EQUIP_TYPE_BOOTS, EQUIP_VALUE_BOOTS_KOKIRI);
+            } else {
+                Inventory_ChangeEquipment(EQUIP_TYPE_BOOTS, bootsValue);
+            }
+            Player_SetEquipmentData(play, player);
+            func_808328EC(player, CUR_EQUIP_VALUE(EQUIP_TYPE_BOOTS) == EQUIP_VALUE_BOOTS_IRON ? NA_SE_PL_WALK_HEAVYBOOTS : NA_SE_PL_CHANGE_ARMS);
+        } else {
+            u16 tunicValue = item - ITEM_TUNIC_KOKIRI + 1;
+            if (CUR_EQUIP_VALUE(EQUIP_TYPE_TUNIC) == tunicValue) {
+                Inventory_ChangeEquipment(EQUIP_TYPE_TUNIC, EQUIP_VALUE_TUNIC_KOKIRI);
+            } else {
+                Inventory_ChangeEquipment(EQUIP_TYPE_TUNIC, tunicValue);
+            }
+            Player_SetEquipmentData(play, player);
+            func_808328EC(player, NA_SE_PL_CHANGE_ARMS);
+        }
+    }
+}
+
+void ClearAssignedTunicsBoots(int32_t unused = 0) {
+    for (int32_t buttonIndex = 0; buttonIndex < 8; buttonIndex++) {
+        int32_t item = gSaveContext.equips.buttonItems[buttonIndex];
+
+        if (item >= ITEM_TUNIC_KOKIRI && item <= ITEM_BOOTS_HOVER) {
+            gSaveContext.equips.buttonItems[buttonIndex] = ITEM_NONE;
+        }
+    }
+}
+
+#define CVAR_TUNICBOOTS_NAME CVAR_ENHANCEMENT("AssignableTunicsAndBoots")
+#define CVAR_TUNICBOOTS_DEFAULT 0
+#define CVAR_TUNICBOOTS_VALUE CVarGetInteger(CVAR_TUNICBOOTS_NAME, CVAR_TUNICBOOTS_DEFAULT)
+
+void RegisterAssignableTunicsBoots() {
+    // make sure we don't change our held/equipped item when changing tunics/boots
+    COND_VB_SHOULD(VB_CHANGE_HELD_ITEM_AND_USE_ITEM, CVAR_TUNICBOOTS_VALUE != CVAR_TUNICBOOTS_DEFAULT, {
+        int32_t item = va_arg(args, int32_t);
+
+        if (item >= ITEM_TUNIC_KOKIRI && item <= ITEM_BOOTS_HOVER) {
+            *should = false;
+        }
+    });
+
+    // make sure we don't crash because tunics/boots don't have assoicated item actions
+    COND_VB_SHOULD(VB_ITEM_ACTION_BE_NONE, CVAR_TUNICBOOTS_VALUE != CVAR_TUNICBOOTS_DEFAULT, {
+        int32_t item = va_arg(args, int32_t);
+
+        if (item >= ITEM_TUNIC_KOKIRI && item <= ITEM_BOOTS_HOVER) {
+            *should = true;
+        }
+    });
+
+    // do something when the player presses a button to use the tunics/boots
+    COND_VB_SHOULD(VB_EXECUTE_PLAYER_ACTION_FUNC, CVAR_TUNICBOOTS_VALUE != CVAR_TUNICBOOTS_DEFAULT, {
+        Player* player = va_arg(args, Player*);
+        PlayState* play = va_arg(args, PlayState*);
```

Comment:
So far we've been externing gPlayState in the files instead, and grabbing the player from there with GET_PLAYER(gPlayState). I'm not actually sure which is preferable, passing along these through the VB arguments or doing it the other way. Wanted to bring it up for other's opinions at the very least.

Reply:
works for me, i'll update it to not use params

Reply:
changed it to use `gPlayState` in https://github.com/HarbourMasters/Shipwright/pull/4978/commits/6a769e99ec38cf46c929a3a9cd189e6455a65bb1